### PR TITLE
pybind/mgr: Rename orchestrator_cli to orchestrator

### PR DESCRIPTION
**Pull request by sebastian-philipp**
Fixes: https://tracker.ceph.com/issues/43674

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by sebastian-philipp** [comment]
- [ ] `--suite rados --filter cephadm`
- [ ] `--suite rados/mgr --filter orchestrator`
- [ ] `--suite rados/mgr --filter dashboard`

**Comment by sebastian-philipp** [comment]
`./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/./obj-x86_64-linux-gnu/CMakeFiles/CMakeTmp/CheckSymbolExists.c:8: undefined reference to `pthread_create'`

**Comment by sebastian-philipp** [comment]
rebased

**Comment by liewegas** [comment]
failures: http://pulpito.ceph.com/sage-2020-01-25_15:18:00-rados-wip-swagner-testing-distro-basic-smithi/

**Comment by sebastian-philipp** [comment]
```
2020-01-25T15:44:20.632 INFO:teuthology.orchestra.run.smithi175.stderr:INFO:cephadm:/usr/bin/ceph:stderr Error ENOTSUP: Module 'ansible' is not enabled (required by command 'orchestrator set backend'): use `ceph mgr module enable ansible` to enable it
2020-01-25T15:44:20.632 INFO:teuthology.orchestra.run.smithi175.stderr:Traceback (most recent call last):
2020-01-25T15:44:20.632 INFO:teuthology.orchestra.run.smithi175.stderr:  File "/home/ubuntu/cephtest/cephadm", line 2846, in <module>
2020-01-25T15:44:20.632 INFO:teuthology.orchestra.run.smithi175.stderr:    r = args.func()
2020-01-25T15:44:20.632 INFO:teuthology.orchestra.run.smithi175.stderr:  File "/home/ubuntu/cephtest/cephadm", line 1861, in command_bootstrap
2020-01-25T15:44:20.632 INFO:teuthology.orchestra.run.smithi175.stderr:    cli(['orchestrator', 'set', 'backend', 'cephadm'])
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:  File "/home/ubuntu/cephtest/cephadm", line 1782, in cli
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:    ).run(timeout=timeout)
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:  File "/home/ubuntu/cephtest/cephadm", line 1555, in run
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:    self.run_cmd(), desc=self.entrypoint, timeout=timeout)
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:  File "/home/ubuntu/cephtest/cephadm", line 497, in call_throws
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:    raise RuntimeError('Failed command: %s' % ' '.join(command))
2020-01-25T15:44:20.633 INFO:teuthology.orchestra.run.smithi175.stderr:RuntimeError: Failed command: /bin/podman run --rm --net=host -e CONTAINER_IMAGE=quay.io/ceph-ci/ceph:3b9e87f714dc9c66b536668a4bd2d774be89b23f -e NODE_NAME=smithi175 -v /var/log/ceph/67d52e26-3f89-11ea-99dd-001a4aab830c:/var/log/ceph:z -v /tmp/ceph-tmpeebgqu46:/etc/ceph/ceph.client.admin.keyring:z -v /tmp/ceph-tmp1s7ke3gw:/etc/ceph/ceph.conf:z --entrypoint /usr/bin/ceph quay.io/ceph-ci/ceph:3b9e87f714dc9c66b536668a4bd2d774be89b23f orchestrator set backend cephadm
2020-01-25T15:44:20.634 DEBUG:teuthology.orchestra.run:got remote process result: 1
```

**Comment by sebastian-philipp** [comment]
> ```
> Error ENOTSUP: Module 'ansible' is not enabled (required by command 'orchestrator set backend'): use `ceph mgr module enable ansible` to enable it
> ```

Turned out that the `orchestrator set backend` prefix is wrongly assigned to the `ansible` module:

https://github.com/ceph/ceph/blob/effe9c9a3f1c299f2f8ff19c98c572aad11c1f00/src/mgr/DaemonServer.cc#L2248-L2259

where

* `py_handler_name == "ansible"`
* `prefix == "orchestrator set backend"`

**Comment by sebastian-philipp** [comment]
fixed!

**Comment by liewegas** [comment]
rebase on top of #33244 ?

**Comment by sebastian-philipp** [comment]
Sure

**Comment by sebastian-philipp** [comment]
http://pulpito.ceph.com:80/swagner-2020-02-17_12:27:04-rados-wip-swagner-testing-distro-basic-smithi/

File: src/pybind/mgr/dashboard/requirements.txt [hunk 1]
```diff
@@ -9,3 +9,5 @@ requests
 Routes
 six
 -e ../../../python-common
+prettytable
+pyyaml
```
Comment (dsavineau) [inline src/pybind/mgr/dashboard/requirements.txt]: @liewegas @sebastian-philipp I think you missed to add those dependencies in the ceph.spec.in file

```console
failed: [mon2 -> mon0] (item=dashboard) => changed=true 
  ansible_loop_var: item
  cmd:
  - ceph
  - --cluster
  - ceph
  - mgr
  - module
  - enable
  - dashboard
  delta: '0:00:00.433621'
  end: '2020-02-18 14:00:06.796205'
  item: dashboard
  msg: non-zero return code
  rc: 2
  start: '2020-02-18 14:00:06.362584'
  stderr: 'Error ENOENT: module ''dashboard'' reports that it cannot run on the active manager daemon: No module named ''yaml'' (pass --force to force enablement)'
  stderr_lines: <omitted>
  stdout: ''
  stdout_lines: <omitted>
```

Comment (sebastian-philipp) [reply src/pybind/mgr/dashboard/requirements.txt]: #33387

**Pull request merged**